skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java

```diff
@@ -367,35 +374,30 @@ public void streamPodLogs(String forUserID, String sessionID, OutputStream out)
     }
     
     public Session getSession(String forUserID, String sessionID) throws Exception {
+        // get session info
         String k8sNamespace = K8SUtil.getWorkloadNamespace();
-        List<String> getSessionCMD = new ArrayList<String>();
-        getSessionCMD.add("kubectl");
-        getSessionCMD.add("get");
-        getSessionCMD.add("--namespace");
-        getSessionCMD.add(k8sNamespace);
-        getSessionCMD.add("pod");
-        getSessionCMD.add("-l");
-        getSessionCMD.add("canfar-net-sessionID=" + sessionID + ",canfar-net-userid=" + forUserID);
-        getSessionCMD.add("--no-headers=true");
-        getSessionCMD.add("-o");
-        
-        String customColumns = "custom-columns=" +
-            "SESSIONID:.metadata.labels.canfar-net-sessionID," + 
-            "USERID:.metadata.labels.canfar-net-userid," +
-            "IMAGE:.spec.containers[0].image," +
-            "TYPE:.metadata.labels.canfar-net-sessionType," +
-            "STATUS:.status.phase," +
-            "NAME:.metadata.labels.canfar-net-sessionName," +
-            "STARTED:.status.startTime," +
-            "DELETION:.metadata.deletionTimestamp";
-        
-        getSessionCMD.add(customColumns);
-                
-        String vncSession = execute(getSessionCMD.toArray(new String[0]));
+        List<String> sessionCMD = getSessionCMD(k8sNamespace, forUserID, sessionID);
+        String vncSession = execute(sessionCMD.toArray(new String[0]));
         log.debug("VNC Session: " + vncSession);
        
         if (StringUtil.hasLength(vncSession)) {
             Session session = constructSession(vncSession.trim());
+            
+            // get expiry time 
+            List<String> sessionExpiryTimeCMD = getSessionExpiryTimeCMD(k8sNamespace, forUserID, sessionID);
+            String sessionExpiryTime = execute(sessionExpiryTimeCMD.toArray(new String[0]));
+            log.debug("Expiry time: " + sessionExpiryTime + " seconds");
+            Instant instant = Instant.parse(session.getStartTime());
+            instant = instant.plus(Integer.parseInt(sessionExpiryTime), ChronoUnit.SECONDS);
```

Comment:
What's the difference between using Instance/ChronoUnit and Date?  Just a newer way of doing it?

Reply:
Most of the methods in Date have been deprecated. Using Instant here only requires two lines of code, so I used it.

Reply:
Okay, thanks, good to know.